ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -41,33 +41,44 @@ const PanelDrawer = () => {
   const { panels, updatePanel } = usePanels();
   const { panelDrawer, closePanelDrawer } = useDashboardApp();
 
+  const panelKey = panelDrawer?.panelKey;
+  const panelDefinition = panelKey ? panels[panelKey] : undefined;
+
   let defaultPanelName = '';
-  let defaultDescription = '';
-  if (panelDrawer?.panelKey) {
+  let defaultPanelDescription = '';
+  let defaultKind = '';
+  let defaultOptions;
+  if (panelDefinition) {
     // editing an existing panel
-    defaultPanelName = panels[panelDrawer.panelKey]?.spec.display.name ?? '';
-    defaultDescription = panels[panelDrawer.panelKey]?.spec.display.description ?? '';
+    defaultPanelName = panelDefinition.spec.display.name ?? '';
+    defaultPanelDescription = panelDefinition.spec.display.description ?? '';
+    defaultKind = panelDefinition.spec.plugin.kind ?? '';
+    defaultOptions = panelDefinition.spec.plugin.spec;
   }
   const [group, setGroup] = useState(panelDrawer?.groupIndex);
   const [panelName, setPanelName] = useState(defaultPanelName);
-  const [panelDescription, setPanelDescription] = useState(defaultDescription);
-  const [kind, setKind] = useState('');
-  const [options, setOptions] = useState<unknown>({});
+  const [panelDescription, setPanelDescription] = useState(defaultPanelDescription);
+  const [kind, setKind] = useState(defaultKind);
+  const [options, setOptions] = useState<unknown>(defaultOptions);
```

Comment:
We'll probably want to rename `option` in light of the change to `spec`, but I'll save that for another day

Reply:
mmm that would be cool to do it asap. Like that the model doesn't diverge :).